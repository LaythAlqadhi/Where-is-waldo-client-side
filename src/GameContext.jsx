import React, { useState, createContext } from 'react';
import PropTypes from 'prop-types';

const initialState = {
  state: { loading: false, error: null },
  status: { isStarted: false, isEnded: true },
  message: null,
};

export const GameContext = createContext(initialState);

export function GameProvider({ children }) {
  const [state, setState] = useState({ loading: false, error: null });
  const [status, setStatus] = useState({
    isStarted: false,
    isEnded: false,
    message: null,
  });
  const [message, setMessage] = useState(null);

  return (
    <GameContext.Provider
      value={{
        state,
        setState,
        status,
        setStatus,
        message,
        setMessage,
      }}
    >
      {children}
    </GameContext.Provider>
  );
}

GameProvider.propTypes = {
  children: PropTypes.node.isRequired,
};
