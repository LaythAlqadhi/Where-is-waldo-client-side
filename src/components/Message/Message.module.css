.message {
  position: fixed;
  top: 100px;
  left: 50%;
  transform: translate(-50%, -50%);
  background-color: darkred;
  padding: 5px 10px;
  border: 1px solid #fff;
  border-radius: 10px;
  font-size: 1.25rem;
  font-weight: 500;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.5);
  z-index: 1000;
}