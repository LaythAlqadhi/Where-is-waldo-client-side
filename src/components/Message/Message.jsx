import React, { useContext } from 'react';
import { GameContext } from '../../GameContext';
import style from './Message.module.css';

const backgroundColor = {
  'Oops! Wrong coordinates.': 'darkred',
  'Nice job! Character found.': 'darkGreen',
  'Congratulations! You Win!': 'darkblue',
};

function Message() {
  const { state, message, setMessage } = useContext(GameContext);

  if (message) {
    setTimeout(() => {
      setMessage(null);
    }, 3000);
  }

  if (state.error) {
    setMessage('Something went wrong.');
  }

  return (
    message && (
      <span
        className={style.message}
        style={{
          backgroundColor: backgroundColor[message],
        }}
      >
        {message}
      </span>
    )
  );
}

export default Message;
