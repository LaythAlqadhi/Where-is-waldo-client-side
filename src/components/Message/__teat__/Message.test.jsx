import React from 'react';
import { render } from '@testing-library/react';
import { vi } from 'vitest';
import { GameContext } from '../../../GameContext';
import Message from '../Message';

it('renders Message component correctly', () => {
  const { asFragment } = render(
    <GameContext.Provider
      value={{
        state: { loading: false, error: null },
        status: { isStarted: true, isEnded: false },
        setMessage: vi.fn(),
        message: 'Nice job! Character found.',
      }}
    >
      <Message />
    </GameContext.Provider>,
  );

  expect(asFragment()).toMatchSnapshot();
});
