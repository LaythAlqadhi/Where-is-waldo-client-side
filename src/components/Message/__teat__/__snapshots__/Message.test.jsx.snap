// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders Message component correctly 1`] = `
<DocumentFragment>
  <span
    class="_message_c69e04"
    style="background-color: darkGreen;"
  >
    Nice job! Character found.
  </span>
</DocumentFragment>
`;
